python/tvm/tir/expr.py

```diff
@@ -562,9 +562,13 @@ def __nonzero__(self):
         return self.value != 0
 
     def __eq__(self, other):
+        if not isinstance(other, IntImm):
+            return False
```

Comment:
this check overlooks the possibility where we want to compare with python's builtin `int`.

if we really want to support the case where we compare `PrimExpr` to `None`, the most conservative check should be:

```suggestion
        if other is None:
            return getattr(self, "handle", None) is None
```

however, i suppose that the TVM FFI should handle the case where IntImm is `IntImm(nullptr)`. Therefore, if it segfaults, it might be more fundamental and the fix I proposed shouldn't be the most proper. Would be great if we take a deeper dive